docs/src/hooks/use-previous.ts

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useRef } from 'react'
```

Comment:
This is interesting. I haven't seen a ref used this way.

Reply:
yeah refs are super powerful and can be used for more than just dom